docs/topics/scripting/custom-script-deps-tutorial.md

```diff
@@ -0,0 +1,332 @@
+                onAnnotations(DependsOn::class, Repository::class, handler = ::configureMavenDepsOnAnnotations)
+            }
+        }
+    )
+   ```
+
+   The `configureMavenDepsOnAnnotations` function is as follows:
+
+   ```kotlin
+    // Handler that reconfigures the compilation on the fly
+    fun configureMavenDepsOnAnnotations(context: ScriptConfigurationRefinementContext): ResultWithDiagnostics<ScriptCompilationConfiguration> {
+        val annotations = context.collectedData?.get(ScriptCollectedData.collectedAnnotations)?.takeIf { it.isNotEmpty() }
+            ?: return context.compilationConfiguration.asSuccess()
+        return runBlocking {
+            resolver.resolveFromScriptSourceAnnotations(annotations)
+        }.onSuccess {
+            context.compilationConfiguration.with { 
+                dependencies.append(JvmDependency(it))
+            }.asSuccess()
+        }
+    }
+    
+    private val resolver = CompoundDependenciesResolver(FileSystemDependenciesResolver(), MavenDependenciesResolver())
+   ```
+
+   You can find the full code [here](https://github.com/Kotlin/kotlin-script-examples/blob/master/jvm/basic/jvm-maven-deps/script/src/main/kotlin/org/jetbrains/kotlin/script/examples/jvm/resolve/maven/scriptDef.kt).
+
+
+On this step, you have defined what developers can write in scripts of this type and how it will be handled.
+The next step is creating the scripting host – the component that handles the script execution.
+
+## Create a scripting host
+
+1. In the scripting host module, add the dependencies in the `dependencies` block of `build.gradle(.kts)`:
+   * Kotlin scripting components that provide the APIs you need for the scripting host
+   * The script definition module you've created previously
+
+   <tabs group="build-script">
+   <tab title="Kotlin" group-key="kotlin">
+
+   ```kotlin
+   dependencies {
+       implementation("org.jetbrains.kotlin:kotlin-scripting-common")
+       implementation("org.jetbrains.kotlin:kotlin-scripting-jvm")
+       implementation("org.jetbrains.kotlin:kotlin-scripting-jvm-host")
+       implementation(project(":script-definition")) // the script definition module
+   }
+   ```
+
+   </tab>
+   <tab title="Groovy" group-key="groovy">
+
+   ```groovy
+   dependencies {
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-common'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-jvm'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-jvm-host'
+       implementation project(':script-definition') // the script definition module
+   }
+   ```
+
+   </tab>
+   </tabs>
+
+2. Create the `src/main/kotlin/` directory in the module and add Kotlin source file, for example, `host.kt`.
+
+3. Define the `main` function for the application. The application should run with one argument – the path to the script file –
+   and execute the script. You'll define the script execution in a separate function `evalFile` in the next step. Declare it empty for now.
```

Comment:
I don't like such sentences "The application should run" because they look complicated to me.

Reply:
Yeah, you got me. I also don't like the word "should" in docs - it doesn't explain who does what.